### PR TITLE
Add Regis to cave

### PR DESCRIPTION
**Pull request by anapple96**
Exactly what the title say, add Regis to cave

### CODE REVIEW

**Comment by Ultima1990** [comment]
Regirock is weaker than the others.

File: src/scripts/dungeons/Dungeon.ts [hunk 1]
```diff
@@ -363,6 +363,17 @@ dungeonList['Sky Pillar'] = new Dungeon('Sky Pillar',
     ],
     40000, 101, 5);
 
+dungeonList['Sealed Cave'] = new Dungeon('Sealed Cave',
+    ['Zubat', 'Poochyena', 'Nosepass'],
```
Comment (RedSparr0w) [inline src/scripts/dungeons/Dungeon.ts:367]: These don't seem to be the correct Pokemon for the cave:
image

File: src/scripts/dungeons/Dungeon.ts [hunk 2]
```diff
@@ -363,6 +363,17 @@ dungeonList['Sky Pillar'] = new Dungeon('Sky Pillar',
...
+    [GameConstants.BattleItemType.xAttack, GameConstants.BattleItemType.Lucky_egg],
+    400000,
+    [
+        new DungeonBossPokemon('Regirock', 1200000, 20),
```
Comment (RedSparr0w) [inline src/scripts/dungeons/Dungeon.ts:371]: Is Regirock supposed to be weaker than the other 2?

**Review by RedSparr0w — CHANGES_REQUESTED** [review]
Thanks for the PR 👍 
Definitely a good idea to get these out of the Roamers Pool.

I believe it should be called the [`Sealed Chamber`](https://bulbapedia.bulbagarden.net/wiki/Sealed_Chamber)

A few extra change will also be needed to make this work:
- The actual map image will need to be updated.
- The dungeon image needs to be added.
- The dungeon needs to be added to the HoennDungeons constant.
- All the statistics for dungeons after this new dungeons index will need to be updated to their new index _(not ideal, but needed)_.

File: src/scripts/dungeons/Dungeon.ts (hunk 2 above)
Comment (anapple96) [reply src/scripts/dungeons/Dungeon.ts]: Fixed

File: src/scripts/dungeons/Dungeon.ts (hunk 1 above)
Comment (anapple96) [reply src/scripts/dungeons/Dungeon.ts]: Fixed

**Comment by anapple96** [comment]
Should be OK

File: src/components/HoennSVG.html [hunk 3]
```diff
@@ -730,6 +730,13 @@
         })
     %>
 
+    <!--Sealed Cave-->
+    <%- include('templates/mapDungeon',
+        { name: "Sealed Cave"
+        , x: 38, y: 35
```
Comment (RedSparr0w) [inline src/components/HoennSVG.html:736]: I think this needs a width/height set too?

File: src/scripts/GameConstants.ts [hunk 4]
```diff
@@ -649,6 +649,7 @@ namespace GameConstants {
     ];
 
     export const HoennDungeons = [
+        'Petalburg Woods',
```
Comment (RedSparr0w) [inline src/scripts/GameConstants.ts:652]: If you are changing the index for this, all the statistics for dungeons between here and the old index will need to be updated.

File: src/scripts/GameConstants.ts [hunk 5]
```diff
@@ -658,10 +659,10 @@ namespace GameConstants {
         'Mt. Pyre',
         'Shoal Cave',
         'Seafloor Cavern',
+        'Sealed Chamber',
```
Comment (RedSparr0w) [inline src/scripts/GameConstants.ts:662]: You will also need to update the statistics for every dungeon index after this one, as it puts them all out of order.

**Review by RedSparr0w — CHANGES_REQUESTED** [review]
Looks good so far, just the update functions that need to be added 👍

**Comment by RedSparr0w** [comment]
I've fixed the update functions, if someone could test them out to make sure I haven't missed something.

Here's the test script I used:
```js
oldValues = {"Viridian Forest":1,"Digletts Cave":2,"Mt. Moon":3,"Rock Tunnel":4,"Power Plant":5,"Pokemon Tower":6,"Seafoam Islands":7,"Pokemon Mansion":8,"Victory Road":9,"Cerulean Cave":10,"Sprout Tower":11,"Ruins of Alph":12,"Union Cave":13,"Slowpoke Well":14,"Ilex Forest":15,"Burned Tower":16,"Tin Tower":17,"Whirl Islands":18,"Mt Mortar":19,"Ice Path":20,"Dark Cave":21,"Mt Silver":22,"Rusturf Tunnel":23,"Granite Cave":24,"Jagged Pass":25,"Fiery Path":26,"Mt. Chimney":27,"Meteor Falls":28,"Mt. Pyre":29,"Shoal Cave":30,"Seafloor Cavern":31,"Cave of Origin":32,"Sky Pillar":33,"Victory Road Hoenn":34,"Petalburg Woods":35,"New Mauville":36,"Oreburgh Gate":37,"Ravaged Path":38,"Eterna Forest":39,"Old Chateau":40,"Wayward Cave":41,"Mt. Coronet South":42,"Iron Island":43,"Mt. Coronet North":44,"Lake Verity":45,"Lake Valor":46,"Lake Acuity":47,"Distortion World":48,"Victory Road Sinnoh":49,"Spear Pillar":50,"Hall of Origin":51,"Fullmoon Island":52,"Newmoon Island":53,"Flower Paradise":54,"Snowpoint Temple":55,"Stark Mountain":56};
newValues= {};
GameConstants.RegionDungeons.flat().forEach((d, i) => {
newValues[d] = App.game.statistics.dungeonsCleared[Statistics.getDungeonIndex(d)]()
})
Object.entries(oldValues).forEach(([d, v]) => {
    if (newValues[d] !== v) console.log(d)
})
```
And the test save from `v0.5.4` attached
_(which is where the `oldValues` is from)_
[[v0.5.4] PokeClickerSave_2020-09-28 09_32_16.txt](https://github.com/pokeclicker/pokeclicker/files/5291441/v0.5.4.PokeClickerSave_2020-09-28.09_32_16.txt)

File: src/scripts/Update.ts [hunk 6]
```diff
@@ -235,6 +235,21 @@ class Update implements Saveable {
...
             }
         }
+        
+        if (this.isOlderVersion(this.saveVersion, '0.5.5')) {
+            try {
+                //Correct statistics
+                saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 22, 34); // Petalburg Woods
+                saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 30, 35); // New Mauville
+                saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 56, 50); // Hall of Origin
+                saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 33); // Hall of Origin
```
Comment (Aegyo) [inline src/scripts/Update.ts:245]: This comment supposed to say sealed chamber?
No other issue as far as I can tell :+1:

**Review by Aegyo — APPROVED** [review]

File: src/scripts/Update.ts (hunk 6 above)
Comment (RedSparr0w) [reply src/scripts/Update.ts:245]: Sure is 🤦 
```suggestion
                saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 33); // Sealed Chamber
```

**Pull request merged**